pkg/rook/client/object.go

```diff
@@ -50,3 +55,100 @@ func (c *RookNetworkRestClient) GetObjectStoreConnectionInfo() (model.ObjectStor
 
 	return connInfo, nil
 }
+
+func (c *RookNetworkRestClient) ListBuckets() ([]model.ObjectBucket, error) {
+	body, err := c.DoGet(path.Join(objectStoreQueryName, bucketsQueryName))
+	if err != nil {
+		return nil, err
+	}
+
+	var buckets []model.ObjectBucket
+	err = json.Unmarshal(body, &buckets)
+	if err != nil {
+		return nil, err
+	}
+
+	return buckets, nil
+}
+
+func (c *RookNetworkRestClient) ListObjectUsers() ([]model.ObjectUser, error) {
+	body, err := c.DoGet(path.Join(objectStoreQueryName, usersQueryName))
+	if err != nil {
+		return nil, err
+	}
+
+	var users []model.ObjectUser
+	err = json.Unmarshal(body, &users)
+	if err != nil {
+		return nil, err
+	}
+
+	return users, nil
+}
+
+func (c *RookNetworkRestClient) GetObjectUser(id string) (model.ObjectUser, error) {
+	body, err := c.DoGet(path.Join(objectStoreQueryName, usersQueryName, id))
+	if err != nil {
+		return model.ObjectUser{}, err
```

Comment:
i think the pattern works best to return `nil` when there is an error rather than instantiating an empty object. It just means you have to change the return type to be a pointer.